src/pykeen/triples/triples_factory.py

```diff
@@ -787,35 +787,28 @@ def merge(self, *others: Self) -> Self:
             mapped_triples.append(other.mapped_triples)
         return self.clone_and_exchange_triples(torch.cat(mapped_triples, dim=0))
 
-    def entities_to_ids(self, entities: Collection[int] | Collection[str]) -> Collection[int]:
+    def _raise_on_string(self, x: int | str, label: str) -> int:
+        if not isinstance(x, int):
+            raise TypeError(f"{self.__class__.__name__} cannot convert {label} IDs from {type(x)} to int.")
+        return x
+
+    def entities_to_ids(self, entities: Iterable[int] | Iterable[str]) -> Sequence[int]:
         """Normalize entities to IDs.
 
-        :param entities: A collection of either integer identifiers for entities or
+        :param entities: A sequence of either integer identifiers for entities or
             string labels for entities (that will get auto-converted)
-        :returns: Integer identifiers for entities
-        :raises ValueError: If the ``entities`` passed are string labels
```

Comment:
it's a bummer the way the docstr-coverage test works that these become "redundant". Maybe we can still save this text?

Reply:
https://github.com/pykeen/pykeen/pull/1521/commits/aa9331272ee6554806140878d1a7eeffd5b95fff

I also changed to `TypeError` which seemed to fit better.

I am also not 100% sure whether the typing really makes sense: The `CoreTriplesFactory` does not support `str` input and that is not covered by the annotation.

Reply:
We could add a generic to the factory to describe that, but maybe there is also a more elegant way? Generally speaking, a subclass should be able to accept more input types (and restrict the output type further, but that is not the case here).

Reply:
good point. Is there an issue if the parent and child class have different type annotations? I think this is called the "liskov substitution principle" and is a pain in some places

Reply:
I just checked what happens if you restrict the type annotation in `CoreTriplesFactory` but that escalates quite a few changes. We also often have these kind of

```python
if isinstance(tf, TriplesFactory):
  ...
else: 
 ...
```
blocks which are not too nice. And now that I think about, it is also quite annoying that all `Dataset` always claim to only have `CoreTriplesFactory`s while most of them actually have `TriplesFactory`s with labels.

Reply:
Maybe something for a different MR though 🙂

Reply:
one of the solutions the "cool kids" say for this is to prefer dependency injection over inheritance.. definitely a discussion for another PR

@mberr btw looks like you're drinking the corporate cool-aid if you're calling them MRs now ;)

Reply:
For requests within a single repo MR might be more appropriate than PR 🤔 😄 

> dependency injection

I have become quite a fan, but I do not want to blindly apply it just for the sake of it as the introduction would probably cause a bit of diff. Another PR sounds good 👍

Reply:
If you are happy with the revived exception part in the docstr, feel free to merge.